muxers/yamux/tests/compliance.rs

```diff
@@ -0,0 +1,38 @@
+use libp2p_yamux::YamuxConfig;
+
+#[async_std::test]
+async fn close_implies_flush() {
```

Comment:
To deduplicate, would it make sense for `libp2p_muxer_test_harness` to offer a `run_all` method, taking a muxer constructor? That would prevent us adding a test to the harness but not to all the muxer implementations.

Reply:
I decided against it because I wanted to have individual tests in the output. Could be solved with a macro I guess but I don't think the current design is too much of a problem to keep in sync and it doesn't involve any magic.